### ISSUE TITLE
CupertinoTextSelectionToolbar does not reflect iOS 16 text toolbar changes

### ISSUE THREAD

**Issue by zbarbuto** [flutter/flutter#113584]
iOS 16 has updated the appearance of the Text Selection Toolbar from the previous black appearance (still used by `CupertinoTextSelectionToolbar`).

Current flutter appearance (Minimal flutter app running on iOS 16.0.3 - just adding a simple `TextField()` to the flutter starter app)

![image](https://user-images.githubusercontent.com/9100419/196305959-a4ad111c-7242-4b44-a8ef-fd6d57b77acf.png)

Previous native appearance (iOS < 16 - screenshot taken from Notes app on iPad running iOS 12)

<img width="495" alt="image" src="https://user-images.githubusercontent.com/9100419/196305666-51f27ade-4270-44d7-bde2-cf82669ce293.png">

New native appearance (iOS 16+ - screenshot taken from Notes app on iPhone running iOS 16.0.3) - light theme with a slight shadow behind it.

![image](https://user-images.githubusercontent.com/9100419/196305935-979428d5-caa6-480d-822d-b004db8ce44d.png)

**Expected results:**

Expected that on iOS 16 the appearance of the text selection toolbar would be grey to match the 'native' selection toolbar.

**Actual results:** 

Toolbar remains black as per iOS < 16 appearance. Also note the tiny grey 'point' to the toolbar arrow.

<details>
<summary>Code sample</summary>

```dart
class MyHomePage extends StatelessWidget {
  @override
  Widget build(BuildContext context) {
    return Scaffold(
      body: Center(
        child: Column(
          mainAxisAlignment: MainAxisAlignment.center,
          children: const <Widget>[
            Padding(padding: EdgeInsets.all(12), child: TextField())
          ],
        ),
      ),
    );
  }
}
```

</details>

<details>
  <summary>Logs</summary>

```
$ flutter doctor -v             
[✓] Flutter (Channel stable, 3.3.4, on macOS 12.6 21G115 darwin-x64, locale en-AU)
    • Flutter version 3.3.4 on channel stable at /Users/zbarbuto/Library/flutter
    • Upstream repository https://github.com/flutter/flutter.git
    • Framework revision eb6d86ee27 (13 days ago), 2022-10-04 22:31:45 -0700
    • Engine revision c08d7d5efc
    • Dart version 2.18.2
    • DevTools version 2.15.0
```

</details>

**Comment by huycozy** [comment]
@zbarbuto 
Thanks for filing the issue. I can reproduce this issue on the latest `stable` and `master` channels.
Tested device: iPhone 14 Pro Max - iOS 16.0

<details>
<summary>Demo (native app & Flutter app)</summary>

| Native app (Reminders) | Flutter app |
| --------------- | --------------- |
<img width="400" src="https://user-images.githubusercontent.com/104349824/196402223-f30d0436-ac2b-482f-b039-4d7e96a35b9f.png"> | <img width="400" src="https://user-images.githubusercontent.com/104349824/196402236-8aebe081-686d-4b6d-bb41-4bf46376f61d.png">

</details>

<details>
<summary>flutter doctor -v (stable and master)</summary>

```bash
[✓] Flutter (Channel stable, 3.3.4, on macOS 12.6 21G115 darwin-x64, locale en-VN)
    • Flutter version 3.3.4 on channel stable at /Users/huynq/Documents/GitHub/flutter
    • Upstream repository https://github.com/flutter/flutter.git
    • Framework revision eb6d86ee27 (23 hours ago), 2022-10-04 22:31:45 -0700
    • Engine revision c08d7d5efc
    • Dart version 2.18.2
    • DevTools version 2.15.0

[✓] Android toolchain - develop for Android devices (Android SDK version 31.0.0)
    • Android SDK at /Users/huynq/Library/Android/sdk
    • Platform android-33, build-tools 31.0.0
    • ANDROID_HOME = /Users/huynq/Library/Android/sdk
    • Java binary at: /Applications/Android Studio.app/Contents/jre/Contents/Home/bin/java
    • Java version OpenJDK Runtime Environment (build 11.0.12+0-b1504.28-7817840)
    • All Android licenses accepted.

[✓] Xcode - develop for iOS and macOS (Xcode 14.0.1)
    • Xcode at /Applications/Xcode14.0.1.app/Contents/Developer
    • Build 14A400
    • CocoaPods version 1.11.3

[✓] Chrome - develop for the web
    • Chrome at /Applications/Google Chrome.app/Contents/MacOS/Google Chrome

[✓] Android Studio (version 2021.2)
    • Android Studio at /Applications/Android Studio.app/Contents
    • Flutter plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/6351-dart
    • Java version OpenJDK Runtime Environment (build 11.0.12+0-b1504.28-7817840)

[✓] IntelliJ IDEA Community Edition (version 2022.2.2)
    • IntelliJ at /Applications/IntelliJ IDEA CE.app
    • Flutter plugin version 70.0.5
    • Dart plugin version 222.4167.21

[✓] IntelliJ IDEA Community Edition (version 2022.1.1)
    • IntelliJ at /Users/huynq/Library/Application Support/JetBrains/Toolbox/apps/IDEA-C/ch-0/221.5591.52/IntelliJ IDEA CE.app
    • Flutter plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/6351-dart

[✓] VS Code (version 1.71.2)
    • VS Code at /Applications/Visual Studio Code.app/Contents
    • Flutter extension version 3.48.0

[✓] Connected device (2 available)
    • macOS (desktop) • macos  • darwin-x64     • macOS 12.6 21G115 darwin-x64
    • Chrome (web)    • chrome • web-javascript • Google Chrome 105.0.5195.125

[✓] HTTP Host Availability
    • All required HTTP hosts are available

• No issues found!

```

```bash
[✓] Flutter (Channel master, 3.5.0-6.0.pre.66, on macOS 12.6 21G115 darwin-x64, locale en-VN)
    • Flutter version 3.5.0-6.0.pre.66 on channel master at /Users/huynq/Documents/GitHub/flutter_master
    • Upstream repository https://github.com/flutter/flutter.git
    • Framework revision 2ea90eaed2 (3 hours ago), 2022-10-17 19:13:49 -0400
    • Engine revision 465ccd2974
    • Dart version 2.19.0 (build 2.19.0-314.0.dev)
    • DevTools version 2.18.0

[✓] Android toolchain - develop for Android devices (Android SDK version 31.0.0)
    • Android SDK at /Users/huynq/Library/Android/sdk
    • Platform android-33, build-tools 31.0.0
    • ANDROID_HOME = /Users/huynq/Library/Android/sdk
    • Java binary at: /Applications/Android Studio.app/Contents/jre/Contents/Home/bin/java
    • Java version OpenJDK Runtime Environment (build 11.0.13+0-b1751.21-8125866)
    • All Android licenses accepted.

[✓] Xcode - develop for iOS and macOS (Xcode 14.0.1)
    • Xcode at /Applications/Xcode14.0.1.app/Contents/Developer
    • Build 14A400
    • CocoaPods version 1.11.3

[✓] Chrome - develop for the web
    • Chrome at /Applications/Google Chrome.app/Contents/MacOS/Google Chrome

[✓] Android Studio (version 2021.3)
    • Android Studio at /Applications/Android Studio.app/Contents
    • Flutter plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/6351-dart
    • Java version OpenJDK Runtime Environment (build 11.0.13+0-b1751.21-8125866)

[✓] IntelliJ IDEA Community Edition (version 2022.2.2)
    • IntelliJ at /Applications/IntelliJ IDEA CE.app
    • Flutter plugin version 70.0.5
    • Dart plugin version 222.4167.21

[✓] IntelliJ IDEA Community Edition (version 2022.1.1)
    • IntelliJ at /Users/huynq/Library/Application Support/JetBrains/Toolbox/apps/IDEA-C/ch-0/221.5591.52/IntelliJ IDEA CE.app
    • Flutter plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/6351-dart

[✓] VS Code (version 1.72.1)
    • VS Code at /Applications/Visual Studio Code.app/Contents
    • Flutter extension version 3.50.0

[✓] Connected device (3 available)
    • SM T225 (mobile) • R9JT3004VRJ • android-arm64  • Android 12 (API 31)
    • macOS (desktop)  • macos       • darwin-x64     • macOS 12.6 21G115 darwin-x64
    • Chrome (web)     • chrome      • web-javascript • Google Chrome 106.0.5249.119

[✓] HTTP Host Availability
    • All required HTTP hosts are available

• No issues found!

```
</details>

**Comment by zbarbuto** [comment]
Worth noting that if the device is set to Dark appearance then the dark selection toolbar is still used.

Screenshot below from the notes app on the same iPhone 12 running iOS 16.0 but in Dark mode.

![thumbnail_Note](https://user-images.githubusercontent.com/9100419/196408014-05d860fe-2665-4d2a-bd4a-b2fe1a1e465d.png)

**Comment by justinmc** [comment]
Closing this as a duplicate of https://github.com/flutter/flutter/issues/111671.  Let me know if I've missed anything from here though.  I'm going to copy over your two light and dark screenshots which are helpful, thank you!

I've got a PR open (https://github.com/flutter/flutter/pull/107193) that cleans up the code behind these toolbars.  I think it won't be too hard to get this looking right after that is merged.
